### PR TITLE
[DPE-1939] Add spark8t library and integration tests

### PR DESCRIPTION
**Pull request by welpaolo**
This PR addresses DPE-1939:

- Add new spark8t library
- Add integration tests

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Spark Image Tests
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+  workflow_call:
+  workflow_dispatch:
```
Comment (deusebio) [inline .github/workflows/integration.yaml]: **niptick** Do we need all of them?

File: rockcraft.yaml [hunk 2]
```diff
@@ -95,38 +95,15 @@ parts:
...
-    organize:
-      spark-client.pyspark: opt/spark/python/bin/spark-client.pyspark
-      spark-client.service-account-registry: opt/spark/python/bin/spark-client.service-account-registry
-      spark-client.spark-shell: opt/spark/python/bin/spark-client.spark-shell
-      spark-client.spark-submit: opt/spark/python/bin/spark-client.spark-submit
-    stage:
-      - opt/spark/python/bin/spark-client.pyspark
-      - opt/spark/python/bin/spark-client.service-account-registry
-      - opt/spark/python/bin/spark-client.spark-shell
-      - opt/spark/python/bin/spark-client.spark-submit
```
Comment (deusebio) [inline rockcraft.yaml]: **question** Why removing this? I believe it was sort of useful to have this short-cut

**Review by deusebio — COMMENTED** [review]
Looks good! Just a couple of questions to address or decide whether to address the comments, and then we can merge. 

Also, given we don't have any python specific project, I'd really switch from tox to Makefile. But we can definitely address this is a separate ticket

**Review by deusebio — COMMENTED** [review]
Looks good! Just a couple of questions to address or decide whether to address the comments, and then we can merge. 

Also, given we don't have any python specific project, I'd really switch from tox to Makefile. But we can definitely address this is a separate ticket

**Review by juditnovak — APPROVED** [review]
LGTM, congrats!

**Review by juditnovak — APPROVED** [review]
LGTM, congrats!

File: .github/workflows/integration.yaml (hunk 1 above)
Comment (welpaolo) [reply .github/workflows/integration.yaml]: No, just left on pull request.

File: rockcraft.yaml (hunk 2 above)
Comment (welpaolo) [reply rockcraft.yaml]: I re-introduce them.

**Comment by deusebio** [comment]
LGTM!

**Review by deusebio — APPROVED** [review]

**Pull request merged**